apps/settings/js/panels/simpin_dialog/simpin_dialog.js

```diff
@@ -0,0 +1,602 @@
+     * for returned results.
+     *
+     * @memberOf SimPinDialog
+     * @param {Object} options
+     * @param {String} options.lockType
+     * @param {String} options.pin
+     * @param {String} options.newPin
+     * @param {String} options.pin2
+     * @param {Boolean} options.enabled
+     * @access private
+     * @return {Promise}
+     */
+    _setCardLock: function(options) {
+      return this._icc.setCardLock(options).then(() => {
+        return true; 
+      }, (error) => {
+        var needToCloseDialog = this._handleCardLockError({
+          lockType: options.lockType,
+          retryCount: error.retryCount
+        });
+        if (needToCloseDialog) {
+          return true;
+        } else {
+          return Promise.reject();
+        }
+      });
+    },
+
+    /**
+     * this function is used when we are going to update fdn contacts under
+     * call panel.
+     *
+     * @memberOf SimPinDialog
+     * @access private
+     * @return {Promise}
+     */
+    _updateFdnContact: function() {
+      //  Updates a FDN contact. For some reason, `icc.updateContact` requires
+      //  the  pin input value instead of delegating to `icc.setCardLock`.
+      //  That means that, in case of failure, the error is different that the
+      //  one that `icc.setCardLock` gives. This means that we have to handle
+      //  it separatedly instead of being able to use the existing
+      //  `handleCardLockError` above.
+      //
+      //  Among other things, it doesn't include the retryCount, so we can't
+      //  tell the user how many remaining tries she has. What a mess.
+      //  This should be solved when bug 1070941 is fixed.
+
+      var fdnContact = this._pinOptions.fdnContact;
+      return this._icc.updateContact('fdn', fdnContact,
+        this._elements.pinInput.value).then(() => {
+          return fdnContact;
+      }, (error) => {
+        switch (error.name) {
+          case 'IncorrectPassword':
+          case 'SimPin2':
+            // TODO: count retries (not supported by the platform) ->
+            // Bug 1070941
+            this._initUI('get_pin2');
+            this._showMessage('fdnErrorMsg');
+            this._elements.pinInput.value = '';
+            this._elements.pinInput.focus();
+            return Promise.reject();
+
+          case 'SimPuk2':
+            this._initUI('unlock_puk2');
+            this._elements.pukInput.focus();
+            return Promise.reject();
+
+          case 'NoFreeRecordFound':
+            alert(l10n.get('fdnNoFDNFreeRecord'));
+            return fdnContact;
+
+          default:
+            console.error('Could not edit FDN contact on SIM card - ', error);
+            return fdnContact;
+        }
+      });
+    },
+
+    /**
+     * We will try to handle all returned errors here when requests are
+     * returned from icc.
+     *
+     * @memberOf SimPinDialog
+     * @param {Object} options
```

Comment:
Please list all options. IMHO there are only two arguments in this case and it is more readable declaring them in the function.  But I have no strong opinion on this.